### ISSUE TITLE
PANIC at dnode.c:715:dnode_reallocate()

### ISSUE THREAD

**Issue by jmesmon** [zfsonlinux/zfs#8574]
### System information
Type | Version/Name
 --- | --- 
Distribution Name	| gentoo
Distribution Version	| 
Linux Kernel	| 5.0.4
Architecture	| amd64
ZFS Version	| 0.8.0-rc3_134_gf72ecb8d
SPL Version	| 0.8.0-rc3_134_gf72ecb8d

### Describe the problem you're observing
Panic during zfs recv

### Describe how to reproduce the problem

I lack specifics here: this was triggered by one of several `zfs recv` invocations by one of a few `zxfer` calls.

### Include any warning/errors/backtraces from the system logs
```
[ 3044.595701] VERIFY3(0 == dnode_set_blksz(dn, blocksize, 0, tx)) failed (0 == 95)
[ 3044.595704] PANIC at dnode.c:715:dnode_reallocate()
[ 3044.595704] Showing stack for process 163276
[ 3044.595707] CPU: 2 PID: 163276 Comm: receive_writer Tainted: P           O      5.0.4-gentoo #1
[ 3044.595707] Hardware name: Gigabyte Technology Co., Ltd. X170-WS ECC/X170-WS ECC-CF, BIOS F20 12/19/2016
[ 3044.595708] Call Trace:
[ 3044.595714]  dump_stack+0x46/0x59
[ 3044.595721]  spl_panic+0xd3/0x106 [spl]
[ 3044.595723]  ? _raw_spin_unlock+0xa/0x12
[ 3044.595746]  ? dbuf_rele_and_unlock+0x3bd/0x3e5 [zfs]
[ 3044.595774]  ? zfs_space_delta_cb+0x12a/0x1dc [zfs]
[ 3044.595776]  ? ___might_sleep+0x34/0x119
[ 3044.595778]  ? _cond_resched+0x20/0x23
[ 3044.595779]  ? _raw_spin_unlock+0xa/0x12
[ 3044.595780]  ? ___might_sleep+0x34/0x119
[ 3044.595782]  ? _raw_spin_unlock+0xa/0x12
[ 3044.595800]  ? dnode_set_blksz+0x22f/0x258 [zfs]
[ 3044.595819]  dnode_reallocate+0x159/0x213 [zfs]
[ 3044.595836]  dmu_object_reclaim_dnsize+0xa4/0xde [zfs]
[ 3044.595854]  receive_object+0x45b/0x794 [zfs]
[ 3044.595874]  receive_writer_thread+0xc5/0x826 [zfs]
[ 3044.595877]  ? _raw_spin_unlock+0xa/0x12
[ 3044.595893]  ? dmu_recv_existing_end+0x7a/0x7a [zfs]
[ 3044.595897]  ? __thread_exit+0x13/0x13 [spl]
[ 3044.595900]  ? thread_generic_wrapper+0x7f/0x87 [spl]
[ 3044.595903]  thread_generic_wrapper+0x7f/0x87 [spl]
[ 3044.595905]  kthread+0xf7/0xfc
[ 3044.595906]  ? kthread_worker_fn+0x17f/0x17f
[ 3044.595908]  ret_from_fork+0x1f/0x40
```

**Comment by edillmann** [comment]
duplicate of #8540

**Comment by tcaputi** [comment]
Closed as duplicate of #8540
